valid.go

```diff
@@ -305,6 +311,9 @@ func (d *decoder) wellformedHead() (t cborType, ai byte, val uint64, err error)
 		}
 		val = uint64(binary.BigEndian.Uint32(d.data[d.off : d.off+4]))
 		d.off += 4
+		if err := d.acceptableFloat(float64(math.Float32frombits(uint32(val)))); err != nil {
+			return 0, 0, 0, err
+		}
```

Comment:
Same as comment above.  We need to check CBOR type before converting `val` to float.